spice/training/actions.py

```diff
@@ -516,20 +522,75 @@ def tokenize_function(examples):
 
         return tokenized_dataset
 
+    def _process_dataset(self, dataset, config, task):
+        # get image processor
+        LOGGGER.info("Loading image processor...")
+        if task in ["train", "test"]:
+            hf_model_repo_id = config["hfModelRepoId"]
+            hf_model_repo_revision = config["hfModelRepoRevision"]
+
+            image_processor = AutoImageProcessor.from_pretrained(
+                pretrained_model_name_or_path=hf_model_repo_id,
+                revision=hf_model_repo_revision,
+                trust_remote_code=True,
+            )
+        elif task == "verify":
+            training_round_number = config["roundNumber"]
+            hf_model_repo_id = config["trainingJob"]["hfModelRepoId"]
+            base_model_repo_id = config["trainingJob"]["baseModelRepoId"]
+            base_model_repo_revision = config["trainingJob"]["baseModelRepoRevision"]
+
+            image_processor = AutoImageProcessor.from_pretrained(
+                pretrained_model_name_or_path=hf_model_repo_id
+                if hf_model_repo_id and training_round_number > 1
+                else base_model_repo_id,
+                revision="main"  # uses main tokenizer
+                if hf_model_repo_id and training_round_number > 1
+                else base_model_repo_revision,
+                trust_remote_code=True,
+            )
```

Comment:
pretty verbose, lets move the ternaries out
```suggestion
            pretrained_model_name_or_path = base_model_repo_id
            revision = "main"
            if hf_model_repo_id and training_round_number > 1:
                pretrained_model_name_or_path = base_model_repo_id
                revision = base_model_repo_revision
            image_processor = AutoImageProcessor.from_pretrained(
                pretrained_model_name_or_path= pretrained_model_name_or_path,
                revision=revison,
                trust_remote_code=True,
            )
```